assets/js/hocs/with-product-variations.js

```diff
@@ -124,7 +124,7 @@ const withProductVariations = createHigherOrderComponent(
 						p.variations &&
 						p.variations.find( ( { id } ) => id === variationId )
 				);
-				return parentProduct[ 0 ].id;
+				return parentProduct.length && parentProduct[ 0 ].id;
```

Comment:
We could simplify it with optional chaining operator

```suggestion
				return parentProduct[ 0 ]?.id;
```